lib/govuk_index/presenters/parts_presenter.rb

```diff
@@ -0,0 +1,58 @@
+module GovukIndex
+  class PartsPresenter
+    def initialize(parts: [])
+      @parts = parts
+    end
+
+    def presented_parts
+      return unless parts && parts.any?
+
+      parts.map do |part|
+        body = body_content(part.fetch("body", [{}]))
+        {
+          "slug" => part["slug"],
+          "title" => part["title"],
+          "body" => body,
+          "summary" => summarise(body),
+        }
+      end
+    end
+
+  private
+
+    attr_reader :parts
+
+    ELLIPSIS = "…".freeze
+    SEPARATOR = " ".freeze
+
+    def body_content(part_bodies)
+      html = html_content(part_bodies)
+      return unless html
+
+      strip_html_tags(html)
+    end
+
+    def summarise(body_content)
+      return unless body_content.present?
+
+      body_content.truncate(75, omission: ELLIPSIS, separator: SEPARATOR)
```

Comment:
Now that this is just a method call on the body content, which is going into the search index, I'm wondering if indexing the summary is useful - could we just truncate the body when presenting search results?  What do you think?

Reply:
Yeah that's interesting. Perhaps we could store the truncated body as the body, and not have to pass a lot of extra data over in search responses? Added bonus that we don't need to re-index!